### PR TITLE
Manually inline copy fn in ig-variants test

### PR DESCRIPTION
**Pull request by mppf**
The inlined function prevents the unordered-forall optimization since
part of the analysis runs before inlining. See #12413.

Reviewed by @ronawho - thanks!

### CODE REVIEW

File: test/studies/bale/indexgather/ig-variants.chpl [hunk 1]
```diff
@@ -56,16 +56,28 @@ proc testit(mode: Mode, param explicit, printStats) {
...
+          unorderedCopy(tmp[i], A[rindex[i]]);
+        else
+          tmp[i] = A[rindex[i]];
+      }
     }
     when Mode.zipArray {
-      forall (t, r) in zip(tmp, rindex) do
-        copy(t, A[r], explicit);
+      forall (t, r) in zip(tmp, rindex) {
+        if explicit then
```
Comment (ronawho) [inline test/studies/bale/indexgather/ig-variants.chpl]: Mind writing these as one-liners to match the promotion case? i.e. 

```chpl
if explicit then unorderedCopy(t, A[r]);
            else t = A[r];
```

Comment (mppf) [reply test/studies/bale/indexgather/ig-variants.chpl]: If I do that, the first one does not fit on one line.
I could do it everywhere else, let the line wrap, or change the promoted case to be multi-line.

Comment (ronawho) [reply test/studies/bale/indexgather/ig-variants.chpl]: I don't mind lines being over 80 chars, but if you want to keep it under let's change the promoted case to be multi-line.

Comment (mppf) [reply test/studies/bale/indexgather/ig-variants.chpl]: OK. I made them multi-line.

**Pull request merged**